File: editor/en/[225]Implement Stack using Queues.js
//leetcode submit region begin(Prohibit modification and deletion)

var MyStack = function () {
    this.queue = [];
};

/**
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function (x) {
    this.queue.unshift(x);
    for(let i = 0; i < this.queue.length; i++) {
        this.queue.unshift(this.queue.pop());
    }
};

/**
 * @return {number}
 */
MyStack.prototype.pop = function () {
    return this.queue.shift();
};

/**
 * @return {number}
 */
MyStack.prototype.top = function () {
    return this.queue[0];
};

/**
 * @return {boolean}
 */
MyStack.prototype.empty = function () {
    return this.queue.length === 0;
};

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[225]Implement Stack using Queues.java
// leetcode submit region begin(Prohibit modification and deletion)
/** Stack implementation using a single queue. */
public class MyStack {
  private final Queue<Integer> queue;

  /** Initialize your data structure here. */
  public MyStack() {
    queue = new LinkedList<>();
  }

  /**
   * Push element x onto stack.
   *
   * @param x the element to push
   */
  public void push(int x) {
    queue.offer(x);
    // Rotate the queue to move the new element to the front
    int sz = queue.size();
    for (int i = 0; i < sz - 1; i++) {
      queue.offer(queue.poll());
    }
  }

  /**
   * Removes the element on top of the stack and returns that element.
   *
   * @return the top element
   */
  public int pop() {
    return queue.isEmpty() ? -1 : queue.poll();
  }

  /**
   * Get the top element.
   *
   * @return the top element without removing it
   */
  public int top() {
    return queue.isEmpty() ? -1 : queue.peek();
  }

  /**
   * Returns whether the stack is empty.
   *
   * @return true if empty, false otherwise
   */
  public boolean empty() {
    return queue.isEmpty();
  }
}
//leetcode submit region end(Prohibit modification and deletion)
